timemachine/fe/mle.py

```diff
@@ -257,14 +257,20 @@ def infer_node_vals_and_errs_networkx(
         the inferred values of `node_val_prop` and `node_stddev_prop`.
     """
     assert isinstance(graph, (nx.DiGraph, nx.MultiDiGraph)), "Graph must be a DiGraph or MultiDiGraph"
+
+    # remove edges without edge_diff, edge_stddev properties
     edges_with_props = [
         e for e, d in graph.edges.items() if d.get(edge_diff_prop) is not None and d.get(edge_stddev_prop) is not None
     ]
     sg = graph.edge_subgraph(edges_with_props).copy()
-
     if not sg.nodes:
         raise ValueError("Empty graph after removing edges without predictions")
 
+    # extract largest connected component
+    connected_components = list(nx.connected_components(sg.to_undirected()))
+    connected_components = sorted(connected_components, key=len, reverse=True)
+    sg = sg.subgraph(connected_components[0])
```

Comment:
nit: avoid list construction
```suggestion
    connected_components = nx.connected_components(sg.to_undirected())
    largest_connected_component = max(connected_components, key=len)
    sg = sg.subgraph(largest_connected_component)
```

Reply:
Good call -- applied in https://github.com/proteneer/timemachine/pull/1483/commits/60af1fe811046cbeaf60bccb7c5638c72ba4d41a .

I also changed the sort key here from `len` to a custom `_sort_key` function in https://github.com/proteneer/timemachine/pull/1483/commits/f9f6938dcfd05a9395f6d0cecd6dbc8b6d9d9fc4 , to break ties using # expt. refs (or node names, as a last resort).